app/scripts/components/common/catalog/controls/hooks/use-filters-with-query.ts

```diff
@@ -32,11 +32,23 @@ export function useFiltersWithURLAtom(): UseFiltersWithQueryResult {
   };
 } 
 
-export function useFiltersWithQS(): UseFiltersWithQueryResult {
-  const navigate = useNavigate();
-  const useQsState = useQsStateCreator({
-    commit: navigate
-  });
+  export function useFiltersWithQS({
+    navigate,
+    push=false
+  }: {
+    navigate: any,
+    push?: boolean,
+  }): UseFiltersWithQueryResult {
+    
+    let navCommit = navigate;
+
+    if (push) {
+      navCommit = ({ search }) => navigate.push(`?${search}`);
```

Comment:
I think there is a potential to make it even more flexible if we just let the instance to pass this whole thing as navigate props. ex. Replace this line: https://github.com/developmentseed/next-veda-ui/pull/3/files#diff-6578b281e50705562cdf8ba73c73ae2bb1df366e18e91e9595106b09cfbf3f89R14 
```
 const controlVars = useFiltersWithQS({navigate: ({ search }) => router.push(`?${search}`)});
```

then let's say if the instance wants to replace the url instead of push, it can handle it on its level. (and no additional push props is needed) What do you think?

Reply:
I think that also makes sense but I think we should work on closing out the PR if this is not blocking we can address this in the routing ticket / make note of this